### PR TITLE
DOCSP-33399 role limitation

### PR DESCRIPTION
**Pull request by kennethdyer**
## Description

Adds sync limitation: mongosync does not sync users and roles.

## Staging

* [General Limitations](https://preview-mongodbkennethdyer.gatsbyjs.io/cluster-sync/DOCSP-33399-role-limitation/reference/limitations/#general-limitations) (last entry)

## JIRA

[DOCSP-33399](https://jira.mongodb.org/browse/DOCSP-33399)

## Build

* [2024-01-26](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=65b4300994771f76570acc82)

### CODE REVIEW

**Review by mmaville-mdb — APPROVED** [review]
LGTM!

File: source/reference/limitations.txt [hunk 1]
```diff
@@ -52,6 +52,7 @@ General Limitations
 - You cannot sync a collection that has a unique index and a non-unique
   index defined on the same field(s).
 - .. include:: /includes/fact-atlas-require-indexes-limitation.rst
+- ``mongosync`` doesn't sync users and roles.
```
Comment (FGasper) [inline source/reference/limitations.txt:55]:
```suggestion
- ``mongosync`` doesn't sync users or roles.
```

**Review by FGasper — CHANGES_REQUESTED** [review]
One quick change to the verbiage seems apropos.

Thanks!

**Review by FGasper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**